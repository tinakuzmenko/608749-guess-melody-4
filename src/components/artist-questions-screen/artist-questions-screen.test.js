import React from "react";
import renderer from "react-test-renderer";
import ArtistQuestionsScreen from "./artist-questions-screen.jsx";

const question = {
  type: `artist`,
  song: {
    artist: `Jim Beam`,
    src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
  },
  answers: [{
    picture: `https://api.adorable.io/avatars/128/${Math.random()}`,
    artist: `John Snow`,
  }, {
    picture: `https://api.adorable.io/avatars/128/${Math.random()}`,
    artist: `Jack Daniels`,
  }, {
    picture: `https://api.adorable.io/avatars/128/${Math.random()}`,
    artist: `Jim Beam`,
  }],
};

describe(`ArtistQuestionsScreen`, () => {
  it(`Renders correctly`, () => {
    const tree = renderer
      .create(<ArtistQuestionsScreen
        question={question} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
