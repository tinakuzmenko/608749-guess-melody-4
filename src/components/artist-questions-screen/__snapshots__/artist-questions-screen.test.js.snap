// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistQuestionsScreen Renders correctly 1`] = `
<section
  className="game game--artist"
>
  <header
    className="game__header"
  >
    <a
      className="game__back"
      href="#"
    >
      <span
        className="visually-hidden"
      >
        Сыграть ещё раз
      </span>
      <img
        alt="Угадай мелодию"
        className="game__logo"
        src="img/melody-logo-ginger.png"
      />
    </a>
    <svg
      className="timer"
      viewBox="0 0 780 780"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        className="timer__line"
        cx="390"
        cy="390"
        r="370"
        style={
          Object {
            "filter": "url(#blur)",
            "transform": "rotate(-90deg) scaleY(-1)",
            "transformOrigin": "center",
          }
        }
      />
    </svg>
    <div
      className="game__mistakes"
    >
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
    </div>
  </header>
  <section
    className="game__screen"
  >
    <h2
      className="game__title"
    >
      Кто исполняет эту песню?
    </h2>
    <div
      className="game__track"
    >
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          type="button"
        />
        <div
          className="track__status"
        >
          <audio
            src="https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg"
          />
        </div>
      </div>
    </div>
    <form
      className="game__artist"
      onSubmit={[Function]}
    >
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="answer-1"
          name="answer"
          type="radio"
          value="artist-1"
        />
        <label
          className="artist__name"
          htmlFor="answer-1"
        >
          <img
            alt="Пелагея"
            className="artist__picture"
            src="http://placehold.it/134x134"
          />
          Пелагея
        </label>
      </div>
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="answer-1"
          name="answer"
          type="radio"
          value="artist-1"
        />
        <label
          className="artist__name"
          htmlFor="answer-1"
        >
          <img
            alt="Пелагея"
            className="artist__picture"
            src="http://placehold.it/134x134"
          />
          Пелагея
        </label>
      </div>
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="answer-1"
          name="answer"
          type="radio"
          value="artist-1"
        />
        <label
          className="artist__name"
          htmlFor="answer-1"
        >
          <img
            alt="Пелагея"
            className="artist__picture"
            src="http://placehold.it/134x134"
          />
          Пелагея
        </label>
      </div>
    </form>
  </section>
</section>
`;
